File: CustomUiComponents/app/src/main/java/com/jarvis/dragdropresearch/funwithshapes/ArcShape.java
package com.jarvis.dragdropresearch.funwithshapes;

import com.jarvis.dragdropresearch.interpolators.AngleInterpolator;

public class ArcShape extends FlashShape {
    private static final int MAX_COMPONENT_COUNT = 2;

    private AngleInterpolator mAngleInterpolator;

    public ArcShape() {
        super(Type.ARC);
    }

    public AngleInterpolator getAngleInterpolator() {
        return mAngleInterpolator;
    }

    public void setAngleInterpolator(
            AngleInterpolator angleInterpolator) {
        mAngleInterpolator = angleInterpolator;
    }

    @Override
    public int getMaxComponents() {
        return MAX_COMPONENT_COUNT;
    }
}

File: CustomUiComponents/app/src/main/java/com/jarvis/dragdropresearch/funwithshapes/ArcShape.kt
package com.jarvis.dragdropresearch.funwithshapes

import com.jarvis.dragdropresearch.interpolators.AngleInterpolator

class ArcShape : FlashShape(Type.ARC) {

    var angleInterpolator:AngleInterpolator? = null

    override fun getMaxComponents(): Int = MAX_COMPONENT_COUNT

    companion object {
        private val MAX_COMPONENT_COUNT: Int = 10
    }
}
